### DISCUSSION TITLE
Transfer 01: Error - Source file file does not exist!

### DISCUSSION THREAD

**Discussion by johnnychoque** [eclipse-edc/Samples#73] [Q&A] [answered]
Hi all,

I am working on Ubuntu 20.04 and I am trying to test the transfer-01-file-transfer sample but when I request the file I get the following error message:

Fatal error while Initiate data flow. Error details: Failed to delegate data transfer to Data Plane: Source file file does not exist!

The following log messages are displayed in the console:

>DEBUG 2023-06-23T12:10:36.79635539 DSP: Incoming TransferRequestMessage for transfer process
DEBUG 2023-06-23T12:10:36.807262667 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state INITIAL
DEBUG 2023-06-23T12:10:37.608320326 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state PROVISIONING
DEBUG 2023-06-23T12:10:37.608652118 TransferProcess: ID c2f9019a-4637-427d-b97e-951b91b00637. Provisioning
DEBUG 2023-06-23T12:10:37.60890685 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state PROVISIONED
DEBUG 2023-06-23T12:10:37.609255328 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state STARTING
DEBUG 2023-06-23T12:10:37.609450844 TransferProcess: ID c2f9019a-4637-427d-b97e-951b91b00637. Initiate data flow
SEVERE 2023-06-23T12:10:37.609740791 TransferProcess: ID c2f9019a-4637-427d-b97e-951b91b00637. Fatal error while Initiate data flow. Error details: Failed to delegate data transfer to Data Plane: Source file file does not exist!
SEVERE 2023-06-23T12:10:37.609815113 Failed to delegate data transfer to Data Plane: Source file file does not exist!
DEBUG 2023-06-23T12:10:37.609932085 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state TERMINATING
DEBUG 2023-06-23T12:10:37.610162945 TransferProcess: ID c2f9019a-4637-427d-b97e-951b91b00637. Send DataRequest to http://localhost:9292/protocol
DEBUG 2023-06-23T12:10:37.626249955 TransferProcess c2f9019a-4637-427d-b97e-951b91b00637 is now in state TERMINATED

I have previously made the following configurations:

1) I have created the provider file to be transferred in **/tmp/provider/test-document.txt** and I have also created the directory where the consumer will save the received file **/tmp/consumer**.

2) I have configured the property edc.samples.transfer.01.asset.path in the provider configuration file transfer/transfer-01-file-transfer/file-transfer-provider/config.properties:
```
edc.samples.transfer.01.asset.path=/tmp/provider
```
3) I have modified the filetransfer.json file as follows:

```json
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@type": "TransferRequestDto",
  "dataDestination": {
    "type": "File",
    "path": "/tmp/consumer",
    "keyName": "keyName"
  },
  "protocol": "dataspace-protocol-http",
  "assetId": "test-document",
  "contractId": "1:test-document:e2cec3f6-abb1-4ffb-b5e2-efc1f49c7648",
  "connectorAddress": "http://localhost:8282/protocol",
  "privateProperties": {},
  "managedResources": false
}
```

Does anyone know how to fix this?

Thanks in advance for the help,
Best regards,

Johnny

**Comment by ndr-brt** [comment, marked as answer]
the property should point to the file, not to a folder, as [stated by the README](https://github.com/eclipse-edc/Samples/tree/main/transfer/transfer-01-file-transfer#provider-connector):
> The property `edc.samples.transfer.01.asset.path` should point to an existing file in our local environment, as this is the file that will be transferred

**Reply by johnnychoque** [reply]
Thank you very much @ndr-brt for helping me to understand the instructions better. Based on that, I have also corrected the path of the destination file in filetransfer.json, in my case it would look like this:
```json
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@type": "TransferRequestDto",
  "dataDestination": {
    "type": "File",
    "path": "/tmp/consumer/test-received.txt",
    "keyName": "keyName"
  },
  "protocol": "dataspace-protocol-http",
  "assetId": "test-document",
  "contractId": "1:test-document:e2cec3f6-abb1-4ffb-b5e2-efc1f49c7648",
  "connectorAddress": "http://localhost:8282/protocol",
  "privateProperties": {},
  "managedResources": false
}
```

**Comment by tujinwei** [comment]
Hi Johnny, 
I have encountered a similar error as you. Could you share how did you solve this bug and make it work to transfer the file?
Thank you!

```
DEBUG 2023-09-06T14:51:08.660251128 DSP: Incoming TransferRequestMessage for transfer process
DEBUG 2023-09-06T14:51:08.661814198 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state INITIAL
DEBUG 2023-09-06T14:51:09.534801276 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state PROVISIONING
DEBUG 2023-09-06T14:51:09.535062585 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Provisioning
DEBUG 2023-09-06T14:51:09.535467287 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state PROVISIONED
DEBUG 2023-09-06T14:51:09.535827835 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state STARTING
DEBUG 2023-09-06T14:51:09.536172188 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Initiate data flow
SEVERE 2023-09-06T14:51:09.536758407 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Fatal error while Initiate data flow. Error details: Failed to delegate data transfer to Data Plane: Source file test-document does not exist!
WARNING 2023-09-06T14:51:09.536840854 Failed to delegate data transfer to Data Plane: Source file test-document does not exist!
DEBUG 2023-09-06T14:51:09.536947087 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state TERMINATING
DEBUG 2023-09-06T14:51:09.537303537 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Send TransferTerminationMessage to http://localhost:9292/protocol
DEBUG 2023-09-06T14:51:09.547859167 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Send TransferTerminationMessage to http://localhost:9292/protocol
SEVERE 2023-09-06T14:51:09.548063587 TransferProcess: ID 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77. Fatal error while Send TransferTerminationMessage to http://localhost:9292/protocol. Error details: {"@type":"dspace:TransferError","dspace:code":"404","dspace:reason":"TransferProcess with correlationId 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 not found","dspace:processId":"2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77","@context":{"dct":"https://purl.org/dc/terms/","edc":"https://w3id.org/edc/v0.0.1/ns/","dcat":"https://www.w3.org/ns/dcat/","odrl":"http://www.w3.org/ns/odrl/2/","dspace":"https://w3id.org/dspace/v0.8/"}}
WARNING 2023-09-06T14:51:09.548187764 {"@type":"dspace:TransferError","dspace:code":"404","dspace:reason":"TransferProcess with correlationId 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 not found","dspace:processId":"2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77","@context":{"dct":"https://purl.org/dc/terms/","edc":"https://w3id.org/edc/v0.0.1/ns/","dcat":"https://www.w3.org/ns/dcat/","odrl":"http://www.w3.org/ns/odrl/2/","dspace":"https://w3id.org/dspace/v0.8/"}}
DEBUG 2023-09-06T14:51:09.548406491 TransferProcess 2194fcc0-9191-4bdb-bb4d-aa9e1dd68b77 is now in state TERMINATED

```

**Reply by tujinwei** [reply]
My JSON file looks like this:
```
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@type": "TransferRequestDto",
  "dataDestination": {
    "type": "File",
    "path": "/tmp/customer/test-received.txt",
    "keyName": "keyName"
  },
  "protocol": "dataspace-protocol-http",
  "assetId": "test-document",
  "contractId": "MQ==:dGVzdC1kb2N1bWVudA==:ZmZjZTY1ZDctOWE2Mi00MGE0LTgzMjEtMjhhZTNhZjY2ODQ4",
  "connectorId": "provider",
  "connectorAddress": "http://localhost:8282/protocol",
  "privateProperties": {},
  "managedResources": false
}
```
And my config like below:
```
web.http.port=8181
web.http.path=/api
web.http.management.port=8182
web.http.management.path=/management
web.http.protocol.port=8282
web.http.protocol.path=/protocol
edc.samples.transfer.01.asset.path=/tmp/provider/test-document.txt
edc.dsp.callback.address=http://localhost:8282/protocol
edc.participant.id=provider
edc.ids.id=urn:connector:provider
ids.webhook.address=http://localhost:8282/protocol

```

**Reply by johnnychoque** [reply]
Hi @tujinwei,

1) Have you created the file to be transferred and the directory of the consumer where it will be stored?

```
/tmp/provider/test-document.txt
/tmp/consumer
```

2) Have you included the file to be transferred in the provider's configuration (samples/transfer/transfer-01-file-transfer/file-transfer-provider/config.properties)?

`edc.samples.transfer.01.asset.path=/tmp/provider/test-document.txt`

**Reply by tujinwei** [reply]
Hi @johnnychoque, 
thanks for the reply, exactly as you said, my problem was I didn't create the consumer folder.
Thank you so much!

Jinwei
